src/lightning_app/storage/mount.py

```diff
@@ -0,0 +1,55 @@
+from dataclasses import dataclass
+from typing import List
+
+__MOUNT_IDENTIFIER__: str = "__mount__"
+__MOUNT_PROTOCOLS__: List[str] = ["s3://"]
+
+
+@dataclass
+class Mount:
+    """
+    Arguments:
+        source: The location which contains the external data which should be mounted in the
+            running work. At the moment, only AWS S3 mounts are supported. This must be a full
+            `s3` style identifier pointing to a bucket and (optionally) prefix to mount. For
+            example: `s3://foo/bar/`.
+
+        root_dir: A fully qualified directory path in the work where external data source should
+            be mounted as a filesystem.
```

Comment:
What does fully qualified mean here? Can we just say it can be a path (relative or absolute). It's probably more important to mention that the path should not already exist, right? And we can check for it and have an error.

Reply:
absolute path. and no, it does not have to exist because it is created by the lighting backend s3 mounting code before the python process starts